DynamicProgramming/Knapsack.java

```diff
@@ -1,39 +1,44 @@
 package DynamicProgramming;
-
+import org.checkerframework.common.value.qual.*;
+import org.checkerframework.checker.index.qual.*;
 /**
  * A DynamicProgramming based solution for 0-1 Knapsack problem
  */
 
 public class Knapsack {
-
-    private static int knapSack(int W, int wt[], int val[], int n) throws IllegalArgumentException {
+    @SuppressWarnings({"cast.unsafe", "array.access.unsafe.high"})
+        /* Cast unsafe arrises whenever the compiler is unable to prove the code statically.*/
+    // array.access.unsafe.high arrises whenever the compiler finds an expression used to index and cannot prove it statically whether the index is valid or not.
+    private static int knapSack(@Positive int W,@Positive int wt[],@Positive @SameLen("#2") int val[], @Positive @SameLen("#2") int n) throws IllegalArgumentException {
         if(wt == null || val == null)
             throw new IllegalArgumentException();
         int i, w;
-        int rv[][] = new int[n + 1][W + 1];    //rv means return value
+        int rv[][] = (int @MinLen(1) [] @MinLen(1) [])new int[n + 1][W + 1];    //rv means return value
 
         // Build table rv[][] in bottom up manner
         for (i = 0; i <= n; i++) {
             for (w = 0; w <= W; w++) {
                 if (i == 0 || w == 0)
-                    rv[i][w] = 0;
-                else if (wt[i - 1] <= w)
-                    rv[i][w] = Math.max(val[i - 1] + rv[i - 1][w - wt[i - 1]], rv[i - 1][w]);
+                    rv[i][(@LTLengthOf("rv[i]") int )w] = 0;
+                else if (wt[(@IndexFor("wt") int)(i - 1)] <= w)
+                    rv[i][(@LTLengthOf("rv[i]") int )w] = Math.max(val[(@IndexFor("val") int)(i - 1)] + rv[i - 1][(@Positive int)(w - wt[(@NonNegative @IndexFor("wt") int)(i - 1)])], rv[i - 1][w]);
                 else
-                    rv[i][w] = rv[i - 1][w];
+                    rv[i][(@LTLengthOf("rv[i]") int )w] = rv[i - 1][w];
             }
         }
 
-        return rv[n][W];
+        return rv[n][(@IndexFor("rv[n]") int)W];
     }
 
-
+    @SuppressWarnings({"assignment.type.incompatible", "cast.unsafe"})
+        /* Cast unsafe arrises whenever the compiler is unable to prove the code statically.*/
+    // array.access.unsafe.high arrises whenever the compiler finds an expression used to index and cannot prove it statically whether the index is valid or not.
     // Driver program to test above function
     public static void main(String args[]) {
-        int val[] = new int[]{50, 100, 130};
-        int wt[] = new int[]{10, 20, 40};
+        @Positive int val @ArrayLen(3) [] = new int[]{50, 100, 130};
+        @Positive int wt @ArrayLen(3) [] = new int[]{10, 20, 40};
         int W = 50;
-        int n = val.length;
-        System.out.println(knapSack(W, wt, val, n));
+        @SameLen("wt") int n = val.length;
```

Comment:
`@SameLen` doesn't make sense on an `int` - it's used to express that two arrays have the same length.

Reply:
I have made this change but I want to know whether there is a way to write some different annotation instead of @SameLen.